### ISSUE TITLE
Fix for Docker secrets syntax still not released

### ISSUE THREAD

**Issue by GrabCAD-Gary** [hadolint/hadolint#1092]
The fix for Docker secrets syntax causing Hadolint to choke on `env=` has still not been released.

```dockerfile
RUN --mount=type=secret,id=SOME_VAR,env=SOME_VAR \
    ...
```

This is rendering hadolint effectively unusable as it can't be used by anyone trying to secure their Docker builds, which is many of us. It also breaks downstream providers such as MegaLinter which use and support Hadolint. We use both, ironically.

The ticket reporting it (#1039) is closed, which it shouldn't be since the fix isn't live.

The release was reported as being 'soon' two months ago and still isn't out!

Requests for updates haven't been answered.

Please get this done, @lorenzo . We need it now.

Many thanks.

**Comment by aholten** [comment]
Trying to adopt Hadolint but this issue, [now ongoing for 9 months](https://docs.docker.com/build/buildkit/dockerfile-release-notes/#1100), renders Hadolint unusable for anyone using mounting build secrets as environment variables since Dockerfile >= 1.10.0.

Issue [#1039](https://github.com/hadolint/hadolint/issues/1039) is closed with the codeowner saying the fix has been made and will be released "soon" (as of March 2025). Wondering if Hadolint is now abandoned, and what other options there are for linting Dockerfiles?

**Comment by dolan-a** [comment]
@purajit -- The latest hadolint release requires the newer version of `language-docker` (v14+), which exists as a git tag and I assume as a haskall package (just not a github release). You can test the two versions of hadolint via:

```sh
docker run --rm -i hadolint/hadolint:v2.12.0 < /my/path/Dockerfile

docker run --rm -i hadolint/hadolint:v2.13.1 < /my/path/Dockerfile
```

**Comment by purajit** [comment]
@dolan-a ah totally my bad, it was just the cache (it was run through the `hadolint-docker` pre-commit check, which uses the `latest` tag, and I discovered I was still using `v2.12.1`). It's all good! Sorry for the noise all.

**Comment by aholten** [comment]
This is now resolved for my team with release v2.13.1, thanks so much!
